### ISSUE TITLE
Is it possible to support LINKTYPE_NFLOG  for pcap file in protocols module

### ISSUE THREAD

**Issue by ligewei** [seladb/PcapPlusPlus#966]
Now, more and more pcap files from Android debugging are basing LINKTYPE_NFLOG, it is very important to support it. 

<img width="566" alt="image" src="https://user-images.githubusercontent.com/11908819/190884996-96b15446-c43c-4c26-b608-7501ab12b768.png">

**Comment by seladb** [comment]
@ligewei thank you for reaching out!

I agree that can be a nice addition to PcapPlusPlus. It's not only about adding `LINKTYPE_NFLOG` but also supporting the Linux Netfilter NFLOG protocol. 

If you or anyone else would consider picking it up that'd be amazing!

Here is a link to a sample pcap file from the Wireshark web-site: https://wiki.wireshark.org/uploads/__moin_import__/attachments/SampleCaptures/nflog.pcap

**Comment by jafar75** [comment]
@seladb If it helps I can to work on this issue

**Comment by jafar75** [comment]
I investigated the examples and I think for now, default linktype is LINKTYPE_ETHERNET to read pcap files.

if I could to distinguish the link type of input, I can manage it to read the desired protocol.
@seladb Did I understand the scope of the issue correctly (completely)?

regards

**Comment by seladb** [comment]
@jafar75 thank you for working on this issue!

The default link type is `LINKTYPE_ETHERNET`, but `PcapFileReaderDevice` supports other link types too. Here is where it reads it from the pcap file:
https://github.com/seladb/PcapPlusPlus/blob/6a864322ac1cbed0188ae5735597d28696ad3f94/Pcap%2B%2B/src/PcapFileDevice.cpp#L265

`PcapNgFileReaderDevice` also supports different link types, but in the pcapng format each packet can have its own link type, so getting it is done per packet:
https://github.com/seladb/PcapPlusPlus/blob/6a864322ac1cbed0188ae5735597d28696ad3f94/Pcap%2B%2B/src/PcapFileDevice.cpp#L387

The link type is read from the pcap/pcapng file and put in the `RawPacket` object, then when parsing the packet [`Packet::createFirstLayer()`](https://github.com/seladb/PcapPlusPlus/blob/6a864322ac1cbed0188ae5735597d28696ad3f94/Packet%2B%2B/src/Packet.cpp#L706) is called and uses the link type to know what should be the first layer of the packet (which matches the link type).

So basically what we need to do here is:
- Create a new layer - `NflogLayer` that parses the NFLOG layer
- NFLOG link type is already [supported](https://github.com/seladb/PcapPlusPlus/blob/6a864322ac1cbed0188ae5735597d28696ad3f94/Packet%2B%2B/src/RawPacket.cpp#L277) so I don't think we need to do anything to get it from the pcap file
- Use the new `NflogLayer` in [`Packet::createFirstLayer()`](https://github.com/seladb/PcapPlusPlus/blob/6a864322ac1cbed0188ae5735597d28696ad3f94/Packet%2B%2B/src/Packet.cpp#L706) to parse the first packet

**Comment by jafar75** [comment]
@seladb thanks, 
I will try to provide a PR for the issue in the next week

**Comment by seladb** [comment]
Hi @jafar75 ! Are you still planning to work on it?

**Comment by jafar75** [comment]
Hi @seladb, 
Sorry for delay, But I'm working on it

**Comment by seladb** [comment]
> Hi @seladb, Sorry for delay, But I'm working on it

Awesome, thank for the update! 🙏

**Comment by seladb** [comment]
Thank you so much @jafar75 for implementing this! 🙏 ❤️ 

The PR was merged to `master` so closing this issue.
